tests/MongoDB.Driver.Core.Tests/Core/Servers/HeartbeatDelayTests.cs

```diff
@@ -38,13 +39,18 @@ public void Dispose_can_be_called_multiple_times()
         public void RequestHeartbeat_should_respect_to_minHeartbeatInterval([Values(10, -1)]int heartbeatIntervalInMinutes)
         {
             var heartbeatInterval = heartbeatIntervalInMinutes == -1 ? Timeout.InfiniteTimeSpan : TimeSpan.FromMinutes(heartbeatIntervalInMinutes);
-            var subject = new HeartbeatDelay(heartbeatInterval, TimeSpan.FromSeconds(2));
-
+            var minHeartbeatInterval = TimeSpan.FromSeconds(2);
+            var stopwatch = Stopwatch.StartNew();
+            var subject = new HeartbeatDelay(heartbeatInterval, minHeartbeatInterval);
             subject.RequestHeartbeat();
-
-            SpinWait.SpinUntil(() => subject.Task.Status != TaskStatus.WaitingForActivation, TimeSpan.FromMilliseconds(1500)).Should().BeFalse();
-            Thread.Sleep(TimeSpan.FromMilliseconds(700));
-            subject.Task.Status.Should().Be(TaskStatus.RanToCompletion);
+            var timeout = TimeSpan.FromMinutes(1);
+            var result = Task.WaitAny(subject.Task, Task.Delay(timeout));
+            if (result != 0)
+            {
+                throw new Exception($"The test timeout {timeout} is exceeded.");
+            }
+            stopwatch.Stop();
+            stopwatch.Elapsed.Should().BeGreaterOrEqualTo(minHeartbeatInterval - TimeSpan.FromMilliseconds(15));
```

Comment:
nit: we like to add blank lines between the Arrange/Act/Assert sections of a test.

Maybe add blank lines after 42 and 52?

Reply:
done